src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx

```diff
@@ -194,23 +201,39 @@ function QueryEditor(props: QueryEditorProps) {
             setLastQueryExecutionSettings(querySettings);
         }
 
+        const queryId = uuidv4();
+        setCurrentQueryId(queryId);
         setResultType(RESULT_TYPES.EXPLAIN);
+
         sendExplainQuery({
             query: input,
             database: tenantName,
             querySettings,
             enableTracingLevel,
+            queryId,
         });
+
         setIsResultLoaded(true);
         props.setShowPreview(false);
         dispatchResultVisibilityState(PaneVisibilityActionTypes.triggerExpand);
     });
 
+    const handleStopButtonClick = React.useCallback(() => {
+        if (currentQueryId) {
+            sendCancelQuery({queryId: currentQueryId, database: tenantName});
+        }
+    }, [currentQueryId, sendCancelQuery, tenantName]);
+
     const handleSendQuery = useEventHandler(() => {
-        if (lastUsedQueryAction === QUERY_ACTIONS.explain) {
-            handleGetExplainQueryClick();
-        } else {
-            handleSendExecuteClick();
+        switch (lastUsedQueryAction) {
+            case QUERY_ACTIONS.explain:
+                handleGetExplainQueryClick();
+                break;
+            case QUERY_ACTIONS.execute:
+                handleSendExecuteClick();
+                break;
+            default:
+                break;
```

Comment:
if `lastUsedQueryAction` is not defined we want to execute the query. And there are only two values, what is the use of the switch here?